### PR TITLE
RHOAIENG-59936: Add Helm bootstrap chart for dashboard module controller

### PR DESCRIPTION
**Pull request by liday-rh**
## Description

Adds a Helm bootstrap chart at `dashboard-operator/charts/dashboard/` for ODH Operator consumption ([RHOAIENG-59936](https://redhat.atlassian.net/browse/RHOAIENG-59936)).

The chart packages **controller bootstrap only** (per the ODH Operator module onboarding guide §3.1):

- **CRD** — `dashboard.opendatahub.io/Dashboard` (synced from `config/crd/bases/`)
- **RBAC** — ServiceAccount, ClusterRole, ClusterRoleBinding
- **Deployment** — dashboard module controller with `RELATED_IMAGE_*` env vars and `--manifests-base-path`
- **ConfigMap** — `opendatahub-dashboard-config` with standalone defaults for platform injection

Dashboard **application** manifests remain embedded in the controller image (`manifests/` → `/opt/manifests/dashboard/`) and are **not** included in this chart.

Also adds Makefile targets (`sync-chart-crds`, `chart-validate`), a CRD drift unit test, and CI Helm validation.

**Follow-up (odh-operator repo):** [RHOAIENG-61029](https://redhat.atlassian.net/browse/RHOAIENG-61029) — register chart in `ODH_COMPONENT_CHARTS` and ModuleHandler.

## How Has This Been Tested?

**Automated:**
- `go test ./...` in `dashboard-operator/` (includes `charts/chart_sync_test.go` CRD sync check)
- CI will run `make chart-validate` (helm lint + template) via updated `dashboard-operator-tests.yml`

**On-cluster (OpenShift 4.21, ROSA):**

Tested the end-to-end integration with [opendatahub-operator PR #3656](https://github.com/opendatahub-io/opendatahub-operator/pull/3656) on a live OpenShift cluster (`api.liday-1.n1ai.p3.openshiftapps.com`).

Setup:
1. Built and ran the odh-operator locally from branch `RHOAIENG-59936` ([PR #3656](https://github.com/opendatahub-io/opendatahub-operator/pull/3656)) using `make run-nowebhook` with `OPERATOR_NAMESPACE=opendatahub DEFAULT_CHARTS_PATH=opt/charts DEFAULT_MANIFESTS_PATH=opt/manifests`
2. The dashboard-operator Helm chart from this PR is already incorporated into the odh-operator repo at `opt/charts/dashboard-operator/`
3. Symlinked `opt/manifests/{hardwareprofiles,osd-configs,connectionAPI,segment}` to their sources under `config/` (normally copied in by the Dockerfile at image build time)
4. Applied CRDs (`oc apply -f config/crd/bases/`), DSCI sample, and DSC sample with `dashboard.managementState: Managed`

Verified:
- [x] Dashboard CR `default-dashboard` created automatically by the module controller, owned by the DSC CR
- [x] Dashboard CR received the gateway domain from GatewayConfig (`rh-ai.apps.rosa.liday-1.n1ai.p3.openshiftapps.com`)
- [x] `dashboard-operator` Deployment created in `opendatahub` namespace via Helm chart rendering
- [x] Deployment includes all 10 `RELATED_IMAGE_*` env vars, leader election args, correct Helm labels (`helm.sh/chart: dashboard-0.1.0`) and platform labels (`platform.opendatahub.io/part-of: dashboard`)
- [x] OwnerReferences set correctly (DSC owns both Dashboard CR and Deployment)
- [x] Supporting resources created: ServiceAccount `dashboard-operator`, ClusterRole `dashboard-operator-role`, ClusterRoleBinding `dashboard-operator-rolebinding`, ConfigMap `opendatahub-dashboard-config`
- [x] Pod in `ImagePullBackOff` — expected since `quay.io/opendatahub/dashboard-operator:latest` image does not exist yet

## Test Impact

- New unit test: `dashboard-operator/charts/chart_sync_test.go` verifies chart CRD matches generated CRD
- CI: Helm install + `make chart-validate` added to dashboard-operator workflow

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes:
- N/A — operator manifests only

After the PR is posted &amp; before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Release Notes

* **New Features**
  * Helm chart deployment for dashboard operator with customizable configuration.
  * Dashboard resources now support component management, module-level configuration, and Perses observability integration.

* **Chores**
  * Dashboard resource API groups have been updated.
  * Added Helm chart validation to CI/CD checks.

### CODE REVIEW

**Comment by liday-rh** [comment]
@coderabbitai review

File: dashboard-operator/charts/dashboard/values.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+image:
+  repository: quay.io/opendatahub/dashboard-operator
+  tag: latest
+  pullPolicy: IfNotPresent
+  pullSecrets: []
+
+manager:
+  replicas: 1
+  leaderElection: true
```
Comment (lucferbux) [inline dashboard-operator/charts/dashboard/values.yaml:24]: **Image name mismatch with Konflux**: This should be `quay.io/opendatahub/odh-dashboard-operator` to match the Konflux ODH build (RHOAIENG-69576). The current name `dashboard-operator` is a different quay repo than what Konflux produces.

In production the module framework overrides this via `ControllerImage` env var injection (see opendatahub-io/opendatahub-operator#3656 review), so it's technically a dev-only default — but keeping them aligned avoids confusion.

```yaml
  repository: quay.io/opendatahub/odh-dashboard-operator
```

File: dashboard-operator/charts/dashboard/values.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# Downstream application images resolved by the module controller when rendering
+# embedded dashboard manifests. Keys match RELATED_IMAGE_* env var names.
+relatedImages:
+  RELATED_IMAGE_ODH_DASHBOARD_IMAGE: quay.io/opendatahub/odh-dashboard:latest
+  RELATED_IMAGE_ODH_MOD_ARCH_MODEL_REGISTRY_IMAGE: quay.io/opendatahub/odh-mod-arch-model-registry:latest
+  RELATED_IMAGE_ODH_MOD_ARCH_GEN_AI_IMAGE: quay.io/opendatahub/odh-mod-arch-gen-ai:latest
+  RELATED_IMAGE_ODH_MOD_ARCH_MLFLOW_IMAGE: quay.io/opendatahub/odh-mod-arch-mlflow:latest
+  RELATED_IMAGE_ODH_MOD_ARCH_MAAS_IMAGE: quay.io/opendatahub/odh-mod-arch-maas:latest
+  RELATED_IMAGE_ODH_MOD_ARCH_EVAL_HUB_IMAGE: quay.io/opendatahub/odh-mod-arch-eval-hub:latest
+  RELATED_IMAGE_ODH_KUBE_RBAC_PROXY_IMAGE: quay.io/opendatahub/odh-kube-rbac-proxy:latest
```
Comment (lucferbux) [inline dashboard-operator/charts/dashboard/values.yaml:63]: **Missing 2 `RELATED_IMAGE` entries**: The ODH operator handler (opendatahub-io/opendatahub-operator#3656 `support.go`) declares 12 env vars but this chart only has 10 defaults. Missing:

```yaml
  RELATED_IMAGE_ODH_AUTOML_IMAGE: quay.io/opendatahub/odh-automl:latest
  RELATED_IMAGE_ODH_AUTORAG_IMAGE: quay.io/opendatahub/odh-autorag:latest
```

These are the pipeline runtime images (distinct from the `MOD_ARCH` UI images). In production they're injected by the operator, but having defaults here helps local chart testing.

File: dashboard-operator/charts/dashboard/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+{{- printf "%sdashboard-operator-rolebinding" .Values.namePrefix }}
+{{- end }}
+
+{{- define "dashboard.configMapName" -}}
+{{- .Values.config.name }}
+{{- end }}
+
+{{- define "dashboard.image" -}}
+{{- printf "%s:%s" .Values.image.repository (.Values.image.tag | default .Chart.AppVersion) }}
+{{- end }}
```
Comment (lucferbux) [inline dashboard-operator/charts/dashboard/templates/_helpers.tpl:73]: **FYI — tag-only format is fine**: This helper uses `printf "%s:%s"` which doesn't support `@sha256:digest` format. That's OK because production image pinning happens at the module framework level (`injectModuleEnv` in the ODH operator directly overwrites `container.image` with the digest-pinned reference, bypassing this template entirely). This helper is only used for dev/local rendering.

No action needed.

File: dashboard-operator/charts/dashboard/templates/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,202 @@
+{{- if .Values.serviceAccount.create }}
```
Comment (lucferbux) [inline dashboard-operator/charts/dashboard/templates/rbac.yaml:1]: **RBAC drift risk**: This is a manual copy of `config/rbac/role.yaml` + `service_account.yaml` + `role_binding.yaml`. Unlike the CRD (which has `make sync-chart-crds` + `chart_sync_test.go`), there's no sync mechanism for RBAC.

Suggestion: Add a test similar to `TestDashboardChartCRDInSync` that validates the ClusterRole rules here match `config/rbac/role.yaml`. Alternatively, add a prominent comment at the top:
```yaml
# IMPORTANT: Keep in sync with config/rbac/role.yaml
```

Fine to defer — but file an issue so it doesn't get forgotten.

File: dashboard-operator/api/v1alpha1/groupversion_info.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// +groupName=dashboard.opendatahub.io
+// +groupName=components.platform.opendatahub.io
```
Comment (lucferbux) [inline dashboard-operator/api/v1alpha1/groupversion_info.go:1]: **Already merged on main**: This change (API group migration to `components.platform.opendatahub.io`) was completed in PR #7795. After rebasing onto `main`, this file should drop out of the diff entirely.

**Review by lucferbux — CHANGES_REQUESTED** [review]
## Review Summary — Helm Chart Bootstrap for Dashboard Module Controller

Hey Liam! Nice work getting the Helm chart scaffolded — the chart structure, CRD sync test, Makefile targets, and CI integration are all solid. I went through this PR alongside the companion PR opendatahub-io/opendatahub-operator#3656 (module handler) and the Konflux onboarding tickets (RHOAIENG-69576/69580) to understand the full deployment chain end-to-end. Below is a summary of what needs to be addressed.

---

### 1. Rebase Needed — API Group Migration Already Merged

The `groupversion_info.go` and `config/rbac/role.yaml` changes in this PR duplicate work already merged in PR #7795. This is why the PR shows as CONFLICTING. The fix is simply to **rebase onto `main`** — those files will resolve cleanly since the target state is identical.

**Action**: Rebase onto `main`. The `groupversion_info.go`, `config/crd/kustomization.yaml`, and `config/rbac/role.yaml` changes should drop out as already-applied.

---

### 2. Singleton Name: `default` → `default-dashboard` (Coordinate with Main)

This PR changes the CEL validation to `self.metadata.name == 'default-dashboard'`, and the CRD in `charts/dashboard/crds/` enforces this. On `main`, the constant is still `DashboardInstanceName = "default"` and the CEL says `'default'`.

The ODH operator repo (`api/components/v1alpha1/dashboard_types.go`) already has `DashboardInstanceName = "default-dashboard"`, so the direction is correct — but this PR also needs to update:

- `dashboard-operator/api/v1alpha1/dashboard_types.go`: change `DashboardInstanceName = "default"` → `"default-dashboard"`
- The kubebuilder CEL marker: `self.metadata.name == 'default-dashboard'`
- Run `make generate && make manifests` so the generated CRD in `config/crd/bases/` matches the chart CRD

Without this, the controller code and the Helm chart CRD disagree on the valid singleton name.

---

### 3. LeaderElectionID Change ✅

The change from `dashboard.opendatahub.io` → `components.platform.opendatahub.io-dashboard` makes sense for API group alignment. Since this is a net-new controller deployment (not upgrading an existing one in-place), no action needed.

---

### 4. Image Name Mismatch with Konflux

The chart defaults to `quay.io/opendatahub/dashboard-operator:latest` but the Konflux CI onboarding tickets build to different image names:
- **ODH (RHOAIENG-69576)**: `quay.io/opendatahub/odh-dashboard-operator`
- **RHOAI (RHOAIENG-69580)**: `quay.io/rhoai/odh-dashboard-operator-rhel9`

The chart default is a dev fallback (production overrides it via the module framework's `ControllerImage` injection — see opendatahub-io/opendatahub-operator#3656 review). But for consistency, the default should match: `quay.io/opendatahub/odh-dashboard-operator`.

Also update `config/manager/kustomization.yaml` to match.

---

### 5. RBAC Drift Risk — Chart vs `config/rbac/`

The chart's `templates/rbac.yaml` is a manual copy of `config/rbac/`. There's no sync mechanism like the CRD has. Over time these will drift.

**Suggestion**: Add a similar sync test for RBAC, or at minimum document the dual-maintenance requirement. Fine to defer with a TODO.

---

### 6. `relatedImages` in values.yaml — Missing 2 Entries

The `values.yaml` lists 10 `RELATED_IMAGE_*` entries but the ODH operator handler (opendatahub-io/opendatahub-operator#3656) injects 12. Missing:
- `RELATED_IMAGE_ODH_AUTOML_IMAGE` (automl pipeline runtime)
- `RELATED_IMAGE_ODH_AUTORAG_IMAGE` (autorag pipeline runtime)

---

### Plan to Address

| # | Issue | Action | Effort |
|---|-------|--------|--------|
| 1 | Rebase conflict | Rebase onto `main` | Low |
| 2 | Singleton name | Update `dashboard_types.go` + CEL marker + regenerate CRD | Low |
| 3 | LeaderElectionID | No action needed ✅ | — |
| 4 | Image name | Update `values.yaml` + `kustomization.yaml` to `odh-dashboard-operator` | Low |
| 5 | RBAC drift | Add sync test or document dual-maintenance | Medium |
| 6 | Missing relatedImages | Add 2 entries to `values.yaml` | Low |

Items 1, 2, 4, and 6 are straightforward and can be done in a single rebase+fixup cycle. See inline comments for specific code locations.

**Comment by liday-rh** [comment]
## Review feedback addressed (4d84108)

Thanks @lucferbux and @coderabbitai — pushed a rebase onto `main` plus a fixup commit addressing the review threads.

### lucferbux
| Item | Status |
|------|--------|
| Rebase onto main | Done — branch rebased; duplicate API-group migration commit resolved against #7795 |
| Singleton name `default-dashboard` | Done — updated `dashboard_types.go`, CEL marker, regenerated CRD, synced chart CRD |
| LeaderElectionID | No change — kept `dashboard.components.platform.opendatahub.io` from main (includes webhook server setup) |
| Image name `odh-dashboard-operator` | Done — `values.yaml` + `config/manager/kustomization.yaml` + Makefile `IMG` |
| RBAC drift | Done — added `TestDashboardChartRBACInSync` (helm-rendered ClusterRole vs `config/rbac/role.yaml`) |
| Missing `RELATED_IMAGE_*` | Done — added `RELATED_IMAGE_ODH_AUTOML_IMAGE` and `RELATED_IMAGE_ODH_AUTORAG_IMAGE` |

### CodeRabbit
| Item | Status |
|------|--------|
| Pin `azure/setup-helm` to commit SHA | Done |
| AGENTS.md CI/CD section | Done — documents Helm install + `make chart-validate` |
| Table-driven chart sync test | Done |
| SA helper `default` fallback | Done — `required()` when `create=false` |
| Image digest helper | No change — per @lucferbux, production pinning is operator-injected |
| RBAC broad permissions | Documented — comment in chart RBAC; permissions required for manifest deployment |
| SA token automount | Done — projected token volume + mount (matches modular-architecture pattern) |
| LeaderElectionID revert | No change — net-new deployment; aligned with main |
| Makefile quoting + `check-chart-crds` | Done |
| Helm install before tests | Done — moved earlier in workflow for RBAC sync test |

Validation: `make lint` and `make test` pass locally.

File: .github/workflows/dashboard-operator-tests.yml [hunk 6]
```diff
@@ -44,6 +44,13 @@ jobs:
         working-directory: dashboard-operator
         run: make test
 
+      - name: Install Helm
+        uses: azure/setup-helm@v4
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply .github/workflows/dashboard-operator-tests.yml]: Fixed in 4d84108: pinned to `azure/setup-helm@1a275c3b69536ee54be43f2070a358922e12c8d4`. Also moved Helm install before `make test` so chart sync tests can use it.

File: dashboard-operator/AGENTS.md [hunk 7]
```diff
@@ -84,6 +90,8 @@ make run              # Run controller locally (requires --namespace and --manif
 # Code generation (run after modifying api/ types)
 make generate         # Generate DeepCopy methods
 make manifests        # Generate CRD YAML from kubebuilder markers
+make sync-chart-crds  # Copy CRD into charts/dashboard/crds
+make chart-validate   # helm lint + template smoke test
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/AGENTS.md:94]: Fixed in 4d84108: CI/CD section now lists Helm install and `make chart-validate` steps.

File: dashboard-operator/charts/chart_sync_test.go [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+	chart, err := os.ReadFile(chartPath)
+	if err != nil {
+		t.Fatalf("read chart CRD: %v", err)
+	}
+
+	if !bytes.Equal(source, chart) {
+		t.Fatalf("chart CRD is out of sync with %s; run 'make sync-chart-crds'", sourcePath)
+	}
+}
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/charts/chart_sync_test.go:174]: Fixed in 4d84108: converted to table-driven format and added `TestDashboardChartRBACInSync`.

File: dashboard-operator/charts/dashboard/templates/_helpers.tpl [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+{{- define "dashboard.deploymentName" -}}
+{{- printf "%sdashboard-operator" .Values.namePrefix }}
+{{- end }}
+
+{{- define "dashboard.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- printf "%s%s" .Values.namePrefix (.Values.serviceAccount.name | default "dashboard-operator") }}
+{{- else }}
+{{- .Values.serviceAccount.name | default "default" }}
+{{- end }}
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/templates/_helpers.tpl:56]: Fixed in 4d84108: `serviceAccount.name` is now required via Helm `required()` when `serviceAccount.create=false`.

File: dashboard-operator/charts/dashboard/templates/_helpers.tpl (hunk 3 above)
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/templates/_helpers.tpl:73]: No change — @lucferbux confirmed digest pinning happens at the ODH operator module framework level; this helper is dev-only.

File: dashboard-operator/charts/dashboard/templates/rbac.yaml [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+      - serviceaccounts
+      - persistentvolumeclaims
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/templates/rbac.yaml:63]: Documented in 4d84108: added sync comment in chart RBAC. Broad mutation verbs are inherent to an operator that deploys embedded manifests containing RBAC/CRD resources — restricting them would break reconciliation.

File: dashboard-operator/charts/dashboard/values.yaml [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+      drop:
+        - ALL
+  podSecurityContext:
+    runAsNonRoot: true
+
+serviceAccount:
+  create: true
+  name: dashboard-operator
+  automountServiceAccountToken: false
+
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/values.yaml:50]: Fixed in 4d84108: added projected `serviceAccountToken` volume + mount to the Deployment (same pattern as modular-architecture).

File: dashboard-operator/cmd/manager/main.go [hunk 12]
```diff
@@ -66,7 +66,7 @@ func main() {
 		Metrics:                metricsserver.Options{BindAddress: metricsAddr},
 		HealthProbeBindAddress: healthProbeAddr,
 		LeaderElection:         leaderElect,
-		LeaderElectionID:       "dashboard.opendatahub.io",
+		LeaderElectionID:       "components.platform.opendatahub.io-dashboard",
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/cmd/manager/main.go]: No change — kept `dashboard.components.platform.opendatahub.io` from main after rebase (includes webhook server). Net-new deployment, no in-place lease migration concern. Aligns with @lucferbux review.

File: dashboard-operator/Makefile [hunk 13]
```diff
@@ -53,6 +53,33 @@ generate: controller-gen ## Generate deepcopy code.
...
+
+CHART_DIR ?= charts/dashboard
+CRD_SOURCE ?= config/crd/bases/components.platform.opendatahub.io_dashboards.yaml
+CRD_CHART ?= $(CHART_DIR)/crds/components.platform.opendatahub.io_dashboards.yaml
+HELM ?= helm
+
+.PHONY: sync-chart-crds
+sync-chart-crds: manifests ## Copy generated CRD into the Helm chart.
+	mkdir -p $(CHART_DIR)/crds
+	cp $(CRD_SOURCE) $(CRD_CHART)
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/Makefile]: Fixed in 4d84108: quoted `$(CHART_DIR)`, `$(CRD_SOURCE)`, and `$(CRD_CHART)` in shell commands.

File: dashboard-operator/Makefile [hunk 14]
```diff
@@ -53,6 +53,33 @@ generate: controller-gen ## Generate deepcopy code.
...
+.PHONY: chart-template
+chart-template: sync-chart-crds ## Render the chart locally for smoke testing.
+	$(HELM) template dashboard $(CHART_DIR) --namespace opendatahub > /dev/null
+
+.PHONY: chart-validate
+chart-validate: chart-lint chart-template ## Lint and render the Helm chart.
+
+.PHONY: check-chart-crds
+check-chart-crds: sync-chart-crds ## Fail if chart CRD differs from generated CRD.
+	@diff -q $(CRD_SOURCE) $(CRD_CHART) >/dev/null || (echo "Chart CRD is out of sync with $(CRD_SOURCE). Run 'make sync-chart-crds'." && exit 1)
```
[the remark this answers is not in the document]
Comment (liday-rh) [reply dashboard-operator/Makefile]: Fixed in 4d84108: removed `sync-chart-crds` prerequisite from `check-chart-crds` so diff detects drift.

File: dashboard-operator/charts/dashboard/values.yaml (hunk 1 above)
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/values.yaml:24]: Fixed in 4d84108: default repository is now `quay.io/opendatahub/odh-dashboard-operator` in `values.yaml`, `config/manager/kustomization.yaml`, and Makefile `IMG`.

File: dashboard-operator/charts/dashboard/values.yaml (hunk 2 above)
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/values.yaml:63]: Fixed in 4d84108: added `RELATED_IMAGE_ODH_AUTOML_IMAGE` and `RELATED_IMAGE_ODH_AUTORAG_IMAGE` pipeline runtime defaults.

File: dashboard-operator/charts/dashboard/templates/_helpers.tpl (hunk 3 above)
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/templates/_helpers.tpl:73]: Acknowledged — no change needed per your note. Production image references are injected by the operator module framework.

File: dashboard-operator/charts/dashboard/templates/rbac.yaml (hunk 4 above)
Comment (liday-rh) [reply dashboard-operator/charts/dashboard/templates/rbac.yaml:1]: Fixed in 4d84108: added `TestDashboardChartRBACInSync` comparing helm-rendered ClusterRole rules against `config/rbac/role.yaml`, plus a sync comment in the template.

File: dashboard-operator/api/v1alpha1/groupversion_info.go (hunk 5 above)
Comment (liday-rh) [reply dashboard-operator/api/v1alpha1/groupversion_info.go:1]: Fixed in 4d84108: rebased onto `main`; this file no longer appears in the PR diff (already merged in #7795).

**Review by lucferbux — APPROVED** [review]
## Re-review: All Issues Addressed ✅

Re-reviewed the full diff (16 files, 1205 lines) after the fix commit. All 6 issues from my [prior review](https://github.com/opendatahub-io/odh-dashboard/pull/8093#pullrequestreview-2967488116) are resolved:

| # | Issue | Status |
|---|-------|--------|
| 1 | **Rebase onto main** | ✅ `groupversion_info.go`, `config/rbac/role.yaml`, `config/crd/kustomization.yaml` dropped from diff |
| 2 | **Singleton name alignment** | ✅ `DashboardInstanceName = "default-dashboard"`, CEL marker updated, CRD regenerated |
| 3 | **LeaderElectionID** | ✅ Already approved (handled separately) |
| 4 | **Image name mismatch** | ✅ `values.yaml` → `quay.io/opendatahub/odh-dashboard-operator`, `kustomization.yaml` updated |
| 5 | **RBAC drift guard** | ✅ Full `TestDashboardChartRBACInSync` test + sync comment in `rbac.yaml` |
| 6 | **Missing relatedImages** | ✅ Both `RELATED_IMAGE_ODH_AUTOML_IMAGE` and `RELATED_IMAGE_ODH_AUTORAG_IMAGE` added (12 total) |

### Additional positives noted
- Helm setup action pinned to SHA (`azure/setup-helm@1a275c3b69536ee54be43f2070a358922e12c8d4`)
- Deployment uses projected SA token with explicit volume mount (security best practice)
- Pod security context: `runAsNonRoot: true`, `allowPrivilegeEscalation: false`, drop ALL capabilities
- `check-chart-crds` Makefile target available as a standalone developer utility
- CRD URL description improved with "last known good" semantics

LGTM — clean, well-structured chart with proper sync guards.

/lgtm
/approve
